dace/transformation/interstate/sdfg_nesting.py

```diff
@@ -1049,3 +1055,17 @@ def apply(self, sdfg):
             outer_state.add_edge(
                 nested_node, val, arrnode, None,
                 memlet.Memlet.from_array(key, arrnode.desc(outer_sdfg)))
+
+        # Remove from the parent SDFG the symbols that are defined in the nested one
+        defined_syms = set()
+
+        for name, desc in nested_sdfg.arrays.items():
+            defined_syms.add(name)
+
+        for e in nested_sdfg.edges():
+            defined_syms |= set(e.data.new_symbols({}).keys())
+
+        defined_syms |= set(nested_sdfg.constants.keys())
+
+        for s in defined_syms:
+            outer_sdfg.symbols.pop(s, None)
```

Comment:
do we add them to the symbols of the nested SDFG a-la nest_state_subgraph? otherwise types can be erased